keras/callbacks.py

```diff
@@ -468,12 +469,55 @@ def _set_model(self, model):
         self.model = model
         self.sess = KTF.get_session()
         if self.histogram_freq and self.merged is None:
-            layers = self.model.layers
-            for layer in layers:
+            for layer in self.model.layers:
+
+                W = None
                 if hasattr(layer, 'W'):
-                    tf.histogram_summary('{}_W'.format(layer.name), layer.W)
+                    W = layer.W
+
+                b = None
                 if hasattr(layer, 'b'):
-                    tf.histogram_summary('{}_b'.format(layer.name), layer.b)
+                    b = layer.b
+
+                weights = []
+                if hasattr(layer, 'trainable_weights'):
+                    weights.extend(layer.trainable_weights)
+                if hasattr(layer, 'non_trainable_weights'):
+                    weights.extend(layer.non_trainable_weights)
+
+                for weight in weights:
+                    #Try and figure out a sensible name
```

Comment:
Fair call; I added this mostly because I had names like Variable_8 for some of my weights.

I sort of wish Keras had something like TF's name scopes which was automatic; I was actually a little surprised when my weights weren't layer_name_1_Variable_8 already.

Happy to remove this and fix the pep8 issues if you think this is mergable otherwise.